client-ts/Microsoft.AspNetCore.SignalR.Test.Server/wwwroot/js/hubConnectionTests.js

```diff
@@ -1,194 +1,149 @@
-                                next: (item) => {
-                                    fail();
-                                },
-                                error: (err) => {
-                                    expect(err.message).toEqual("An error occurred.");
-                                    done();
-                                },
-                                complete: () => {
-                                    fail();
-                                }
-                            });
-
-                    })
-                    .catch(e => {
-                        fail(e);
-                        done();
+            it('rethrows an exception from the server when streaming', function (done) {
+                var errorMessage = "An error occurred.";
+
+                var logger = new signalR.ConsoleLogger(signalR.LogLevel.Information);
+                var hubConnection = new signalR.HubConnection(new signalR.HttpConnection(TESTHUBENDPOINT_URL, { transport: transportType, logger: logger }), logger, protocol);
+
+                hubConnection.start().then(function () {
+                    hubConnection.stream('ThrowException', errorMessage).subscribe({
+                        next: function next(item) {
+                            fail();
+                        },
+                        error: function error(err) {
+                            expect(err.message).toEqual("An error occurred.");
+                            done();
+                        },
+                        complete: function complete() {
+                            fail();
+                        }
                     });
+                }).catch(function (e) {
+                    fail(e);
+                    done();
+                });
             });
 
-            it(`can receive server calls`, done => {
-                let logger = new signalR.ConsoleLogger(signalR.LogLevel.Information);
-                let hubConnection = new signalR.HubConnection(new signalR.HttpConnection(
-                    TESTHUBENDPOINT_URL,
-                    { transport: transportType, logger: logger }),
-                    logger,
-                    protocol);
+            it('can receive server calls', function (done) {
+                var logger = new signalR.ConsoleLogger(signalR.LogLevel.Information);
+                var hubConnection = new signalR.HubConnection(new signalR.HttpConnection(TESTHUBENDPOINT_URL, { transport: transportType, logger: logger }), logger, protocol);
 
-                const message = "你好 SignalR！";
+                var message = "你好 SignalR！";
 
-                let callbackPromise = new Promise((resolve, reject) => {
-                    hubConnection.on("Message", msg => {
-                        expect(msg).toBe(message);
-                        resolve();
-                    });
+                hubConnection.on("Message", function (msg) {
+                    expect(msg).toBe(message);
```

Comment:
So we are no longer guaranteed to be asserting this?

Reply:
We are - the test won't complete without invoking this callback (unless start/invoke throws which will cause a test failure). If the callback is not invoked within the timeout (5 secs by default) the test will fail.